pkg/models/commandvalue.go

```diff
@@ -30,632 +26,593 @@ const (
-		str = cv.stringValue
-		return
-	}
-
-	reader := bytes.NewReader(cv.NumericValue)
-
-	switch cv.Type {
-	case v2.ValueTypeBool:
-		var res bool
-		err := binary.Read(reader, binary.BigEndian, &res)
-		if err != nil {
-			str = err.Error()
-		}
-		str = strconv.FormatBool(res)
-	case v2.ValueTypeUint8:
-		var res uint8
-		err := binary.Read(reader, binary.BigEndian, &res)
-		if err != nil {
-			str = err.Error()
-		}
-		str = strconv.FormatUint(uint64(res), 10)
-	case v2.ValueTypeUint16:
-		var res uint16
-		err := binary.Read(reader, binary.BigEndian, &res)
-		if err != nil {
-			str = err.Error()
-		}
-		str = strconv.FormatUint(uint64(res), 10)
-	case v2.ValueTypeUint32:
-		var res uint32
-		err := binary.Read(reader, binary.BigEndian, &res)
-		if err != nil {
-			str = err.Error()
-		}
-		str = strconv.FormatUint(uint64(res), 10)
-	case v2.ValueTypeUint64:
-		var res uint64
-		err := binary.Read(reader, binary.BigEndian, &res)
-		if err != nil {
-			str = err.Error()
-		}
-		str = strconv.FormatUint(res, 10)
-	case v2.ValueTypeInt8:
-		var res int8
-		err := binary.Read(reader, binary.BigEndian, &res)
-		if err != nil {
-			str = err.Error()
-		}
-		str = strconv.FormatInt(int64(res), 10)
-	case v2.ValueTypeInt16:
-		var res int16
-		err := binary.Read(reader, binary.BigEndian, &res)
-		if err != nil {
-			str = err.Error()
-		}
-		str = strconv.FormatInt(int64(res), 10)
-	case v2.ValueTypeInt32:
-		var res int32
-		err := binary.Read(reader, binary.BigEndian, &res)
-		if err != nil {
-			str = err.Error()
-		}
-		str = strconv.FormatInt(int64(res), 10)
-	case v2.ValueTypeInt64:
-		var res int64
-		err := binary.Read(reader, binary.BigEndian, &res)
-		if err != nil {
-			str = err.Error()
-		}
-		str = strconv.FormatInt(res, 10)
-	case v2.ValueTypeFloat32:
-		var res float32
-		binary.Read(reader, binary.BigEndian, &res)
-		str = fmt.Sprintf("%e", res)
-	case v2.ValueTypeFloat64:
-		var res float64
-		binary.Read(reader, binary.BigEndian, &res)
-		str = fmt.Sprintf("%e", res)
-	case v2.ValueTypeBinary:
-		// produce string representation of first 20 bytes of binary value
-		str = fmt.Sprintf(fmt.Sprintf("Binary: [%v...]", string(cv.BinValue[:20])))
-	default:
-		// ArrayType
-		str = cv.stringValue
-	}
-
-	return
+func (cv *CommandValue) ValueToString() string {
+	return fmt.Sprintf("%v", cv.Value)
 }
 
 // String returns a string representation of a CommandValue instance.
-func (cv *CommandValue) String() (str string) {
-	originStr := fmt.Sprintf("Origin: %d, ", cv.Origin)
-	valueStr := cv.Type + ": " + cv.ValueToString()
-	str = originStr + valueStr
-
-	return
+func (cv *CommandValue) String() string {
+	return fmt.Sprintf("DeviceResource: %s, %s: %s", cv.DeviceResourceName, cv.Type, cv.ValueToString())
 }
 
 // BoolValue returns the value in bool data type, and returns error if the Type is not Bool.
-func (cv *CommandValue) BoolValue() (bool, error) {
+func (cv *CommandValue) BoolValue() (bool, errors.EdgeX) {
```

Comment:
Why/where do we need these functions that return the value ??

Reply:
this is useful to get the value in `HandleWriteCommand` in `ProtocolDriver`
(or at least device-simple is using it)

Reply:
Humm, do we use these in any other devices services?

Reply:
I verified that we do use these helper functions in device services like SNMP, so I am good with leaving them.